### PR TITLE
FEAT: use halo2-ecc's ecdsa implementation [Ready for review]

### PR DESCRIPTION
**Pull request by silathdiir**

### CODE REVIEW

**Comment by lispc** [comment]
@kunxian-xia do you need to update TxCircuit::min_num_rows_block in this PR?

File: Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,11 @@ funty = { git = "https://github.com/ferrilab/funty/", rev = "7ef0d890fbcd8b3def1
 
 [patch."https://github.com/privacy-scaling-explorations/halo2.git"]
 halo2_proofs = { git = "https://github.com/scroll-tech/halo2.git", branch = "scroll-dev-1220" }
+# halo2_proofs = { path = "../halo2/halo2_proofs" }
 
+#[patch."https://github.com/scroll-tech/halo2.git"]
+# halo2_proofs = { git = "https://github.com/scroll-tech/halo2.git", branch = "scroll-dev-1220" }
+# halo2_proofs = { path = "../halo2/halo2_proofs" }
```
Comment (lispc) [inline Cargo.toml:26]: we don't need this comments

**Comment by kunxian-xia** [comment]
> @kunxian-xia do you need to update TxCircuit::min_num_rows_block in this PR?

I think this function does not need modification.

**Comment by lispc** [comment]
@zhenfeizhang @kunxian-xia the reason why CI failed is that in halo2-ecc we enabled ark-std::timer by default. This timer seesm not works well in CI. (cpu type? i don't know the root reason). So i disabled timer by default in halo2-ecc ( and added a feature flag to enable/disable it).

https://github.com/scroll-tech/halo2-ecc/commit/83826a6a5fbec9cd02b97c2f202da3e05da2a306

**Pull request merged**